gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherApi.java

```diff
@@ -84,97 +80,80 @@
       listTopicSubscriptionsIterableCallable;
   private final ApiCallable<DeleteTopicRequest, Empty> deleteTopicCallable;
 
-  public static class ResourceNames {
-
-    // =======================
-    // ResourceNames Constants
-    // =======================
-
-    /**
-     * A PathTemplate representing the fully-qualified path to represent
-     * a project resource.
-     *
-     * <!-- manual edit -->
-     * <!-- end manual edit -->
-     */
-    private static final PathTemplate PROJECT_PATH_TEMPLATE =
-        PathTemplate.create("projects/{project}");
-
-    /**
-     * A PathTemplate representing the fully-qualified path to represent
-     * a topic resource.
-     *
-     * <!-- manual edit -->
-     * <!-- end manual edit -->
-     */
-    private static final PathTemplate TOPIC_PATH_TEMPLATE =
-        PathTemplate.create("projects/{project}/topics/{topic}");
-
-    private ResourceNames() {}
-
-    // ==============================
-    // Resource Name Helper Functions
-    // ==============================
-
-    /**
-     * Formats a string containing the fully-qualified path to represent
-     * a project resource.
-     *
-     * <!-- manual edit -->
-     * <!-- end manual edit -->
-     */
-    public static final String formatProjectPath(String project) {
-      return PROJECT_PATH_TEMPLATE.instantiate("project", project);
-    }
+  /**
+   * A PathTemplate representing the fully-qualified path to represent
+   * a project resource.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  private static final PathTemplate PROJECT_PATH_TEMPLATE =
+      PathTemplate.create("projects/{project}");
 
-    /**
-     * Formats a string containing the fully-qualified path to represent
-     * a topic resource.
-     *
-     * <!-- manual edit -->
-     * <!-- end manual edit -->
-     */
-    public static final String formatTopicPath(String project, String topic) {
-      return TOPIC_PATH_TEMPLATE.instantiate("project", project, "topic", topic);
-    }
+  /**
+   * A PathTemplate representing the fully-qualified path to represent
+   * a topic resource.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  private static final PathTemplate TOPIC_PATH_TEMPLATE =
+      PathTemplate.create("projects/{project}/topics/{topic}");
 
-    /**
-     * Parses the project from the given fully-qualified path which
-     * represents a project resource.
-     *
-     * <!-- manual edit -->
-     * <!-- end manual edit -->
-     */
-    public static final String parseProjectFromProjectPath(String projectPath) {
-      return PROJECT_PATH_TEMPLATE.parse(projectPath).get("project");
-    }
+  /**
+   * Formats a string containing the fully-qualified path to represent
+   * a project resource.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static final String formatProjectName(String project) {
+    return PROJECT_PATH_TEMPLATE.instantiate("project", project);
```

Comment:
The path template string that we get as input, which is language independent and relates to the definition of the API in the proto, supports a different set of things than what MessageFormat supports: https://github.com/googleapis/gax-java/blob/master/src/main/java/com/google/api/gax/protobuf/PathTemplate.java